client/src/app/core/repositories/base-repository.ts

```diff
@@ -340,11 +340,20 @@ export abstract class BaseRepository<V extends BaseViewModel, M extends BaseMode
         this.repositoryServiceCollector.errorService.showError(errorMessage);
     };
 
-    protected sendActionToBackend(actionType: ActionType, payload: any): Promise<any> {
-        return this.actions.sendRequest(actionType, payload).catch(this.raiseError);
+    protected async sendActionToBackend(actionType: ActionType, payload: any): Promise<any> {
+        return await this.handleRequest(this.actions.sendRequest(actionType, payload));
     }
 
-    protected sendBulkActionToBackend(actionType: ActionType, payload: any[]): Promise<any> {
-        return this.sendBulkActionToBackend(actionType, payload).catch(this.raiseError);
+    protected async sendBulkActionToBackend(actionType: ActionType, payload: any[]): Promise<any> {
+        return await this.handleRequest(this.actions.sendBulkRequest(actionType, payload));
+    }
+
+    protected async handleRequest(request: Promise<any>): Promise<any> {
```

Comment:
This can be done in raise error -> Wrap `this.actions.send*Request` into `this.raiseError` -> Unify both methods.